### PR TITLE
Add mmap DBGet microbench parameters

### PR DESCRIPTION
**Pull request by ajkr**
I tried evaluating #9611 using DBGet microbenchmarks but mostly found the change is well within the noise even for hundreds of repetitions; meanwhile, the InternalKeyComparator CPU it saves is 1-2% according to perf so it should be measurable. In this PR I tried adding a mmap mode that will bypass compression/checksum/block cache/file read to focus more on the block lookup paths, and also increased the Get() count.

### CODE REVIEW

File: microbench/db_basic_bench.cc [hunk 1]
```diff
@@ -596,36 +597,27 @@ static void DBGet(benchmark::State& state) {
...
-    for (int64_t max_data : {128l << 20, 512l << 20}) {
-      for (int64_t per_key_size : {256, 1024}) {
-        for (bool enable_statistics : {false, true}) {
-          for (bool negative_query : {false, true}) {
-            for (bool enable_filter : {false, true}) {
-              b->Args({comp_style, max_data, per_key_size, enable_statistics,
-                       negative_query, enable_filter});
-            }
-          }
+  for (int64_t max_data : {128l << 20, 512l << 20}) {
```
Comment (riversand963) [inline microbench/db_basic_bench.cc:611]: It's a little surprising to see that compaction style has little effect on read performance.

**Review by riversand963 — APPROVED** [review]
LGTM. Will let @jay-zhuang determine if there is anything I have missed.

File: microbench/db_basic_bench.cc (hunk 1 above)
Comment (ajkr) [reply microbench/db_basic_bench.cc:611]: The 128MB didn't do compaction. The 512MB setting did a few compactions. Compaction style might make a difference in the latter case. But I thought it'd only catch read performance changes at an arbitrary early point in populating an LSM, which didn't feel necessary to measure. I can add it back though if anyone wants.

Comment (ajkr) [reply microbench/db_basic_bench.cc]: Added back the removed parameters.

**Review by jay-zhuang — APPROVED** [review]
LGTM

**Pull request closed without merging**